src/AutoRest.CSharp/Common/Input/InputTypes/InputEnumTypeValue.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace AutoRest.CSharp.Common.Input;
 
-internal record InputEnumTypeValue(string Name, object Value, string? Summary, string? Doc)
+internal abstract record InputEnumTypeValue(string Name, object Value, string? Summary, string? Doc)
```

Comment:
think we should take this change to MTG to avoid instantiation of this.

Reply:
yep

Reply:
and to clarify, I think "using the input library from MTG" should not be part of our plan - it will be a huge change and we might encounter various of issues. At least in short time, we do not need to consider to use the types from MTG - having these types defined in this library actually has its advantages.